lib/SPIRV/OCL20ToSPIRV.cpp

```diff
@@ -1594,6 +1594,29 @@ void OCL20ToSPIRV::visitCallKernelQuery(CallInst *CI, StringRef DemangledName) {
                  /*BuiltinFuncMangleInfo*/ nullptr, &Attrs);
 }
 
+// Add postfix to overloaded intel subgroup block read/write builtins
+// so new functions can be distinguished.
+static void processSubgroupBlockReadWriteINTEL(CallInst *CI,
+                                               OCLBuiltinTransInfo &Info,
+                                               const Type *DataTy, Module *M) {
+  unsigned VectorNumElements = 1;
+  if (DataTy->isVectorTy())
+    VectorNumElements = DataTy->getVectorNumElements();
+  unsigned ElementBitSize = DataTy->getScalarSizeInBits();
+  Info.Postfix = "_";
+  Info.Postfix +=
+      getIntelSubgroupBlockDataPostfix(ElementBitSize, VectorNumElements);
```

Comment:
That will generate double underscore like `__us2` in `_Z41__spirv_SubgroupImageBlockWriteINTEL__us214ocl_image2d_woDv2_iDv2_s`.
Is it intentional? If not we can
```suggestion
  Info.Postfix =
      getIntelSubgroupBlockDataPostfix(ElementBitSize, VectorNumElements);
```
Maybe we don't need underscores at all?

Reply:
Everything that comes after underscore considered to be just postfix for internal functions created between ocl2tospirv pass and writer, and it won't be visible for others. In writer, this postfix is dropped without any analysis.
However, if there is no underscore postfix it will be an error. Extra underscore solves this problem for integer subgroup_block builtins where no underscore will be generated by postfix function and doesn't do wrong things for other versions. In case we have "intel_sub_group_block_write2" without extra "_" there will be generated something like "__spirv_SubgroupImageBlockWriteINTEL2" and then "dePrefixSPIRVName" will not delete "2" from name (since it searches for underscore) and translation will be incorrect. If we want to drop underscores, "dePrefixSPIRVName" should be changed too. But this can complicate analysis of real spirv op name. It will be quite problematic to find spirv name without any markers like underscore.